crates/rust-analyzer/src/handlers/request.rs

```diff
@@ -1846,7 +1844,10 @@ pub(crate) fn handle_call_hierarchy_outgoing(
                 .into_iter()
                 // This is the range relative to the caller
                 .filter(|it| it.file_id == fpos.file_id)
-                .map(|it| to_proto::range(&line_index, it.range))
+                .filter_map(|it| {
+                    let line_index = snap.file_line_index(it.file_id).ok()?;
```

Comment:
This is always the file index for `fpos.file_id` so we can just hoist this out of the for loop `let line_index = snap.file_line_index(fpos.file_id)`

Reply:
I didn't notice that we had filtered out other files.